### PR TITLE
Turn off colors when the shell it not interactive

### PR DESCRIPTION
**Pull request by danyspin97**
Allow overide by using SYSTEMD_COLORS.

### CODE REVIEW

File: sdbootutil [hunk 1]
```diff
@@ -28,7 +28,12 @@ image=
 
 color_red=
 color_end=
-if [ "$SYSTEMD_COLORS" != "false" ] && [ "$SYSTEMD_COLORS" != "0" ]; then
+# The shell is interactive and the colors haven't been disabled forcefully
+# or the shell is not interactive but the colors have been forcefully enabled
+if { [[ $- == *i* ]] \
+    && [ "$SYSTEMD_COLORS" != "false" ] && [ "$SYSTEMD_COLORS" != "0" ]; } \
+    || [[ $- != *i* ]] \
+    && { [ "$SYSTEMD_COLORS" == "true" ] || [ "$SYSTEMD_COLORS" == "1" ]; }; then
```
Comment (aplanas) [inline sdbootutil:36]: LGTM but the style so far is use `=` instead of `==` , and maybe `[[ "$-" =~ "i" ]]`

Comment (danyspin97) [reply sdbootutil:36]: `[[ "$-" =~ "i" ]]` triggered a warning with shellcheck and it suggested to use `=` instead.

File: sdbootutil [hunk 2]
```diff
@@ -28,7 +28,12 @@ image=
 
 color_red=
 color_end=
-if [ "$SYSTEMD_COLORS" != "false" ] && [ "$SYSTEMD_COLORS" != "0" ]; then
+# The shell is interactive and the colors haven't been disabled forcefully
+# or the shell is not interactive but the colors have been forcefully enabled
+if { [[ "$-" = *i* ]] \
+    && [ "$SYSTEMD_COLORS" != "false" ] && [ "$SYSTEMD_COLORS" != "0" ]; } \
+    || [[ $- != *i* ]] \
```
Comment (aplanas) [inline sdbootutil:35]: Missing quotes here?

File: sdbootutil (hunk 1 above)
Comment (aplanas) [reply sdbootutil]: Very interesting, do you have the ID?

Comment (aplanas) [reply sdbootutil]: oh never mind, I found it

Comment (aplanas) [reply sdbootutil]: If I read correctly the suggestion is to use `[[ "$-" =~ i ]]` ?

Comment (danyspin97) [reply sdbootutil]: Updated the PR with the correct version.

File: sdbootutil (hunk 2 above)
Comment (danyspin97) [reply sdbootutil]: Yea, they are not really needed but they don't hurt either.

File: sdbootutil [hunk 3]
```diff
@@ -28,7 +28,12 @@ image=
 
 color_red=
 color_end=
-if [ "$SYSTEMD_COLORS" != "false" ] && [ "$SYSTEMD_COLORS" != "0" ]; then
+# The shell is interactive and the colors haven't been disabled forcefully
+# or the shell is not interactive but the colors have been forcefully enabled
+if { [[ "$-" =~ i ]] \
```
Comment (aplanas) [inline sdbootutil:33]: Ah Danilo, very sorry to waste your time.

After reading from you that shellcheck complained I changed the sdbootutil =~ operators to match the expectations. I did it here:

https://github.com/openSUSE/sdbootutil/pull/111/commits/01a56be188ef7bf570ddece106d5bbadfce9aa39

So following your initial change, I moved to `[[ "$-" == *"i"* ]]` as recommended in https://www.shellcheck.net/wiki/SC2076

Comment (danyspin97) [reply sdbootutil:33]: I see, I will update the commit as well then.

**Review by aplanas — APPROVED** [review]

**Pull request merged**